packages/legend-studio/src/stores/sidebar-state/WorkflowsState.ts

```diff
@@ -437,3 +459,70 @@ export class WorkspaceWorkflowsState {
     }
   }
 }
+
+export class WorkspaceWorkflowsState extends WorkflowsState {
+  constructor(editorStore: EditorStore, sdlcState: EditorSDLCState) {
+    super(editorStore, sdlcState);
+    makeObservable(this, {
+      editorStore: false,
+      sdlcState: false,
+      fetchWorkflowsState: observable,
+      logState: observable,
+      workflowStates: observable,
+      fetchAllWorkflows: flow,
+    });
+  }
+
+  get label(): string {
+    return 'WORKSPACE WORKFLOWS';
+  }
+
+  override get workspace(): Workspace {
+    return this.editorStore.sdlcState.activeWorkspace;
+  }
+}
+
+export class ProjectVersionWorkflowsState extends WorkflowsState {
+  _version: Version | undefined;
+  constructor(
+    editorStore: EditorStore,
+    sdlcState: EditorSDLCState,
+    version: Version | undefined,
+  ) {
+    super(editorStore, sdlcState);
+    makeObservable(this, {
+      editorStore: false,
+      sdlcState: false,
+      fetchWorkflowsState: observable,
+      logState: observable,
+      workflowStates: observable,
+      fetchAllWorkflows: flow,
+    });
+    this._version = version;
+  }
+
+  get label(): string {
+    return `PROJECT WORKFLOWS ${this.version ? `VERSION ${this.version}` : ''}`;
+  }
+
+  override get version(): string | undefined {
+    return this._version?.id.id;
+  }
+}
+
+export class UnsupportedWorkflowsState extends WorkflowsState {
```

Comment:
This is not a bad idea, but why don't we just do `workflowState?: WorkflowState`. This seems quite overkill to me...